driver/compenv.ml

```diff
@@ -641,13 +641,18 @@ let process_action
   | ProcessCFile name ->
       readenv ppf (Before_compile name);
       Location.input_name := name;
-      let obj_name = match !output_name with
-        | None -> c_object_of_filename name
-        | Some n -> n
-      in
-      if Ccomp.compile_file ~output:obj_name name <> 0
```

Comment:
An alternative implementation would be to keep the definition of `obj_name` and replace the line above with
```ocaml
Ccomp.compile_file ?output:!output_name
```